oct-memcached-preprovision.go

```diff
@@ -1,162 +1,157 @@
-	NumCacheNodes:        aws.Int64(numcachenodes),
-	Port:                 aws.Int64(6379),
-	SecurityGroupIds: []*string{
-		aws.String(os.Getenv("ELASTICACHE_SECURITY_GROUP")),
-	},
-	Tags: []*elasticache.Tag{
-		{
-			Key:   aws.String("Name"),
-			Value: aws.String(name),
+	svc := elasticache.New(session.New(&aws.Config{
+		Region: aws.String(os.Getenv("REGION")),
+	}))
+
+	params := &elasticache.CreateCacheClusterInput{
+		CacheClusterId:          aws.String(name),
+		AutoMinorVersionUpgrade: aws.Bool(true),
+		CacheNodeType:           aws.String(cachenodetype),
+		CacheParameterGroupName: aws.String(cacheparametergroupname),
+		CacheSubnetGroupName:    aws.String("memcached-subnet-group"),
+		Engine:                  aws.String("Memcached"),
+		NumCacheNodes:           aws.Int64(numcachenodes),
+		Port:                    aws.Int64(6379),
+		SecurityGroupIds: []*string{
+			aws.String(os.Getenv("ELASTICACHE_SECURITY_GROUP")),
+		},
+		Tags: []*elasticache.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(name),
+			},
+			{
+				Key:   aws.String("billingcode"),
+				Value: aws.String(billingcode),
+			},
 		},
-                {
-                        Key:   aws.String("billingcode"),
-                        Value: aws.String(billingcode),
-                },
-	},
-      }
-    resp, err := svc.CreateCacheCluster(params)
-
-    if err != nil {
-	fmt.Println(err.Error())
-	return err.Error()
-    }
-
-    fmt.Println(resp)
-    return name
+	}
+	resp, err := svc.CreateCacheCluster(params)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		return err.Error()
+	}
+
+	fmt.Println(resp)
+	return name
 }
 
-func insertnew(name string, plan string, claimed string){
-     uri:=os.Getenv("BROKER_DB")
-     db, err := sql.Open("postgres", uri)
-        if err != nil {
-                  fmt.Println(err)
-                  os.Exit(2)
-        }
-     var newname string
-     err = db.QueryRow("INSERT INTO provision(name,plan,claimed) VALUES($1,$2,$3) returning name;", name, plan, claimed).Scan(&newname)
-
-        if err != nil {
-                  fmt.Println(err)
-                  os.Exit(2)
-        }
-     fmt.Println(newname)
-     err =db.Close()
+func insertnew(name string, plan string, claimed string) {
+	uri := os.Getenv("BROKER_DB")
+	db, err := sql.Open("postgres", uri)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(2)
+	}
+	var newname string
+	err = db.QueryRow("INSERT INTO provision(name,plan,claimed) VALUES($1,$2,$3) returning name;", name, plan, claimed).Scan(&newname)
+
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(2)
+	}
+	fmt.Println(newname)
+	err = db.Close()
 }
 
-func main(){
+func main() {
 
-     uri:=os.Getenv("BROKER_DB")
-     db, err := sql.Open("postgres", uri)
+	uri := os.Getenv("BROKER_DB")
+	db, err := sql.Open("postgres", uri)
 	if err != nil {
-                  fmt.Println(err)
-                  os.Exit(2)
+		fmt.Println(err)
+		os.Exit(2)
 	}
 
```

Comment:
Same thing about defer.Close()but remove line 155 add add `defer db.Close()` to line 110.